graph-builder/src/registry.rs

```diff
@@ -98,28 +101,29 @@ pub fn fetch_releases(
     let mut thread_runtime = tokio::runtime::current_thread::Runtime::new()?;
 
     let registry_host = trim_protocol(&registry);
+    let login_scope = format!("repository:{}:pull", &repo);
 
-    let mut client = dkregistry::v2::Client::configure(&Core::new()?.handle())
+    let client = dkregistry::v2::Client::configure(&Core::new()?.handle())
         .registry(registry_host)
         .insecure_registry(false)
         .username(username.map(|s| s.to_string()))
         .password(password.map(|s| s.to_string()))
         .build()
-        .map_err(|e| format_err!("{}", e))?;
-
-    let authenticated_client = thread_runtime
-        .block_on(authenticate_client(
-            &mut client,
-            format!("repository:{}:pull", &repo),
-        ))
-        .map_err(|e| format_err!("{}", e))?;
+        .map_err(|e| format_err!("{}", e));
 
-    let releases = get_tags(repo.to_owned(), authenticated_client)
-        .and_then(|tag| get_manifest_and_layers(tag, repo.to_owned(), authenticated_client))
-        .and_then(|(tag, layer_digests)| {
+    let releases = future::result(client)
```

Comment:
How about creating or refactoring a function which consumes the `client` and incorporates some of the following `and_then`'s, thus abstracting some of the future specific mechanics here? It would also help readability by giving the functionality a name.

Reply:
I'm fine with leaving this chain as-is for now. We'll need to revisit this later to fit caching and parallel fetches anyway.

Reply:
Is there a timely need for this PR? I for one don't see that so we can wait until there's time to revisit this PR.

Reply:
I'm experimenting with making the scanning logic asynchronous and encapsulated in actix actors, which should be easier to parallelize. Removing the `block_on` as done in this PR is a pre-requisite for that, and a good cleanup in general.